File: CodingBat/Python/Warmup-2/string_splosion.py
"""
Given a non-empty string like "Code" return a string like "CCoCodCode".

string_splosion('Code') → 'CCoCodCode'
string_splosion('abc') → 'aababc'
string_splosion('ab') → 'aab'

@author unobatbayar

"""
def string_splosion(str):
  test = ""
  for x in range(1, len(str)-2):
    test = test + str[:x+2]
  if len(str) == 1:
    return str
  elif len(str) == 2:
    return str[0] + str
  return str[:1]*2 + str[1:2] + test + str

File: CodingBat/Java/Warmup-2/stringSplosion.java
/**
Given a non-empty string like "Code" return a string like "CCoCodCode".

stringSplosion("Code") → "CCoCodCode"
stringSplosion("abc") → "aababc"
stringSplosion("ab") → "aab"

@author unobatbayar
**/

public String stringSplosion(String str) {
  String result = "";
  for(int i = 0; i <str.length(); i++){
    result = str.substring(0, str.length()-i) + result;
  }
  return result;
}
